src/ReTestItems.jl

```diff
@@ -214,10 +224,19 @@ function _runtests(shouldrun, paths, nworkers::Int, nworker_threads::Int, worker
             return _runtests_in_current_env(shouldrun, paths, proj_file, nworkers, nworker_threads, worker_init_expr, testitem_timeout, retries, verbose_results, debug, report, logs)
         else
             @debugv 1 "Activating test environment for `$proj_file`"
-            return Pkg.activate(proj_file) do
-                TestEnv.activate() do
-                    _runtests_in_current_env(shouldrun, paths, proj_file, nworkers, nworker_threads, worker_init_expr, testitem_timeout, retries, verbose_results, debug, report, logs)
+            orig_proj = Base.active_project()
+            try
+                if haskey(TEST_ENVS, proj_file) && isfile(TEST_ENVS[proj_file])
+                    testenv = TEST_ENVS[proj_file]
+                    Base.set_active_project(testenv)
```

Comment:
Should we also be checking that the orig_proj hasn't been changed at all? I think we might want to keep a timestamp in the cache as well, and check that the project is still the same as it was before, otherwise this might fail because we need to `instantiate` the test-only project env again.

Reply:
yeah, probably a good idea.